src/test/docker/docker-run-checks.sh

```diff
@@ -59,9 +59,13 @@ Options:\n\
 "
 
 # check if running in OSX
-if [[ "$(uname)" == "Darwin" ]]; then
+if [[ "$(uname)" == "Darwin" ]] && [[ $FORCE_GNU_GETOPT != 1 ]]; then
     # BSD getopt
-    GETOPTS=`getopt $short_opts -- $*`
+    GETOPTS=`getopt $short_opts -- $(echo $* | sed "s/=/ /g")`
+    export EXTRA_MACOS_STUFF="\n\
+    You are using BSD getopt on macOS. If you have gnu-getopt in your PATH, \n\
+    force the script to use that by setting FORCE_GNU_GETOPT=1.\n"
+    declare usage=${usage}${EXTRA_MACOS_STUFF}
```

Comment:
I don't think you need to use `declare` again here since `usage` has already been declared. In fact, if this were in a function it might declare `usage` as `local` and the modifications would be lost, so I would remove use of `declare` here.

Also, I thought this issue was about BSD getopt not supporting `=`. Should that be mentioned in the extra help message? i.e. something like "either avoid using '=' between options and values or force the use of GNU getopt with FORCE_GNU_GETOPT=1"?

Reply:
> "either avoid using '=' between options and values

This message isn't necessary with the proposed changes which strip out all `=` in the `GETOPT` variable (line 65) which is effectively all `=` signs before the `--`. I can add it if you still want it but I just wanted to make sure you didn't miss line 65:

```
GETOPTS=`getopt $short_opts -- $(echo $* | sed "s/=/ /g")`
```

Reply:
So I guess this PR title needs to be updated because it's not just a help message change anymore.

Reply:
Oh, and the commit message too. Whoops.

Reply:
Ok ignore (basically) everything I said above. This should be addressed now.